pybop/costs/pybamm_cost.py

```diff
@@ -0,0 +1,41 @@
+import pybamm
+
+
+class PybammCost:
+    @staticmethod
+    def variable_name():
+        raise NotImplementedError()
+
+    @staticmethod
+    def variable_expression():
+        raise NotImplementedError()
+
+    @staticmethod
+    def parameters() -> [(pybamm.Parameter, float)]:
+        raise NotImplementedError()
+
+    def add_to_model(self, model: pybamm.BaseModel, param: pybamm.ParameterValues):
+        model.variables[self.variable_name()] = self.variable_expression()
+        for parameter, default_value in self.parameters():
+            param.update({parameter: default_value}, check_already_exists=False)
+
+
+class PybammSumSquaredError(PybammCost):
+    """
+    A SumSquaredError cost implementation within Pybamm.
+    """
+
+    @staticmethod
+    def variable_name():
+        return "SumSquaredError"
+
+    @staticmethod
+    def variable_expression():
+        return pybamm.ExplicitTimeIntegral(
```

Comment:
you want to use `pybamm.DiscreteTimeData` and `pybamm.DiscreteTimeSum`, see the test linked below for an example of use:
https://github.com/pybamm-team/PyBaMM/blob/cb17226225a236ee542664c353d1a7a3219ae39d/tests/unit/test_expression_tree/test_unary_operators.py#L762